azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -318,18 +333,15 @@ public void cleanUpDanglingYarnApplications() {
     try {
       logger.info("cleanUpDanglingYarnApplications start ");
 
-      Set<Integer> executionStoppedFlows = getExecutionStoppedFlows();
-      logger.info("Get executionStoppedFlows: " +
-          executionStoppedFlows.stream().map(Object::toString).collect(Collectors.joining(",")));
-
+      Set<Integer> recentTerminationFlows = getRecentTerminationFlows();
       // get those flows terminated but containers are still alive (failed to properly killed)
       Set<Integer> toBeCleanedContainers = getContainersOfTerminatedFlows();
       logger.info("Get terminatedContainers: " +
           toBeCleanedContainers.stream().map(Object::toString).collect(Collectors.joining(",")));
 
       // combine both sets
-      toBeCleanedContainers.addAll(executionStoppedFlows);
-      logger.info("Get the set of all executions: " +
+      toBeCleanedContainers.addAll(recentTerminationFlows);
+      logger.info("The whole set of all executions to clean yarn apps: " +
```

Comment:
What's the expected/observed number of flows with 'dangling' yarn applications and is this change likely to have a performance impact on the overall working of cleanup-manager (single thread monitoring for this and dispatching for cleanup) ?

Reply:
Will investigate

Reply:
Investigated for this question and below are evidences:
1. The number of "recently terminated flows" doesn't increase dramatically - I checked the busy clusters' metric graph, and run mysql queries, each batch of "recently terminated flows" should be < 100, and averagely 1 running yarn app per flow;
2. Run the mysql queries used by cleanupManager directly and the performance is good enough, each finish with 1.5s;
3. In previous test, the GetApplications() api call can easily handle input of ~200 execution_IDs and no performance issue;
4. In previous test, killing ~100 yarn application in a batch works well with the current parallel killing method (using `FixedThreadPool`);